scalafix-testkit/src/main/scala/scalafix/testkit/RuleTest.scala

```diff
@@ -31,36 +30,48 @@ object RuleTest {
       val input = test.toInput
       val dialect = args.scalaVersion.dialect(args.sourceScalaVersion)
       val tree = dialect(input).parse[Source].get
-      val comment = SemanticRuleSuite.findTestkitComment(tree.tokens)
-      val syntax = comment.syntax.stripPrefix("/*").stripSuffix("*/")
-      val conf = Conf.parseString(test.testName, syntax).get
-      val scalafixConfig = conf.as[ScalafixConfig].get
-      val doc = v1.SyntacticDocument(
-        tree.pos.input,
-        LazyValue.now(tree),
-        DiffDisable.empty,
-        scalafixConfig
-      )
-      val sdoc =
-        v1.SemanticDocument.fromPath(
-          doc,
-          test.semanticdbPath,
-          classLoader,
-          symtab
-        )
-      val decoderSettings =
-        RuleDecoder.Settings().withConfig(scalafixConfig)
-      val decoder = RuleDecoder.decoder(decoderSettings)
-      val rulesConf = ConfGet
-        .getKey(conf, "rules" :: "rule" :: Nil)
-        .getOrElse(Conf.Lst(Nil))
-      val config = Configuration()
-        .withConf(conf)
-        .withScalaVersion(args.scalaVersion.value)
-        .withScalacOptions(args.scalacOptions)
-        .withScalacClasspath(args.classpath.entries)
-      val rules = decoder.read(rulesConf).get.withConfiguration(config).get
-      (rules, sdoc)
+      val allComments = SemanticRuleSuite.filterPossibleTestkitComments(tree.tokens)
+      val lazilyParsedComments = allComments.view.map{ comment =>
+        val syntax = comment.syntax.stripPrefix("/*").stripSuffix("*/")
+        Conf.parseString(test.testName, syntax) match {
+          case Configured.Ok(conf) =>
+            val scalafixConfig = conf.as[ScalafixConfig].get
+            val doc = v1.SyntacticDocument(
+              tree.pos.input,
+              LazyValue.now(tree),
+              DiffDisable.empty,
+              scalafixConfig
+            )
+            val sdoc =
+              v1.SemanticDocument.fromPath(
+                doc,
+                test.semanticdbPath,
+                classLoader,
+                symtab
+              )
+            val decoderSettings =
+              RuleDecoder.Settings().withConfig(scalafixConfig)
+            Some((RuleDecoder.decoder(decoderSettings), conf, sdoc))
+          case Configured.NotOk(_) => None
+        }
+      }
+
+      lazilyParsedComments.collectFirst {
+        case Some((decoder, conf, sdoc)) =>
+          val rulesConf = ConfGet
+            .getKey(conf, "rules" :: "rule" :: Nil)
+            .getOrElse(Conf.Lst(Nil))
+          val config = Configuration()
+            .withConf(conf)
+            .withScalaVersion(args.scalaVersion.value)
+            .withScalacOptions(args.scalacOptions)
+            .withScalacClasspath(args.classpath.entries)
+          val rules = decoder.read(rulesConf).get.withConfiguration(config).get
+          (rules, sdoc)
+      } match {
+        case None =>
```

Comment:
it looks like you forgot to unpack the happy path

Reply:
The idea is that we ignore the unhappy path, i.e. we only care about the first success case which occurs on `collectFirst`.

Or am I missing something?